tests/connections/test_unix_socket_connection.py

```diff
@@ -68,37 +69,60 @@ def tearDown(self):
         os.unlink(self.socketname)
 
     def test_unix_socket_connection_connect_read(self):
-        self.connection = UnixSocketConnection(
+        connection = UnixSocketConnection(
             path=self.socketname, timeout=DEFAULT_TIMEOUT
         )
-        self.connection.connect()
-        self.connection.read()
-        self.connection.disconnect()
+        connection.connect()
+        connection.read()
+        connection.disconnect()
 
     def test_unix_socket_connection_connect_send_bytes_read(self):
-        self.connection = UnixSocketConnection(
+        connection = UnixSocketConnection(
             path=self.socketname, timeout=DEFAULT_TIMEOUT
         )
-        self.connection.connect()
-        self.connection.send(bytes("<gmp/>", 'utf-8'))
-        self.connection.read()
-        self.connection.disconnect()
+        connection.connect()
+        connection.send(bytes("<gmp/>", 'utf-8'))
+        connection.read()
+        connection.disconnect()
 
     def test_unix_socket_connection_connect_send_str_read(self):
-        self.connection = UnixSocketConnection(
+        connection = UnixSocketConnection(
             path=self.socketname, timeout=DEFAULT_TIMEOUT
         )
-        self.connection.connect()
-        self.connection.send("<gmp/>")
-        self.connection.read()
-        self.connection.disconnect()
+        connection.connect()
+        connection.send("<gmp/>")
+        connection.read()
+        connection.disconnect()
+
+    def test_unix_socket_connect_file_not_found(self):
+        connection = UnixSocketConnection(path="foo", timeout=DEFAULT_TIMEOUT)
+        with self.assertRaises(GvmError) as e:
+            connection.connect()
+            self.assertEqual(str(e), 'Socket foo does not exist')
```

Comment:
assertRaises can get a second parameter for testing the error message directly. see https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises

Reply:
I have seen it, but was not sure about using it in a context_manager ...!